src/test/java/org/mule/test/http/functional/AbstractHttpTlsRevocationTestCase.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.test.http.functional;
+
+import static java.security.cert.CRLReason.KEY_COMPROMISE;
+import static org.apache.commons.lang3.exception.ExceptionUtils.getRootCause;
+import static org.hamcrest.CoreMatchers.instanceOf;
+import static org.hamcrest.core.Is.is;
+import static org.hamcrest.core.IsEqual.equalTo;
+import static org.junit.Assert.assertThat;
+import static org.mule.functional.junit4.matchers.MessageMatchers.hasPayload;
+
+import org.mule.functional.junit4.MuleArtifactFunctionalTestCase;
+import org.mule.runtime.core.api.event.CoreEvent;
+import org.mule.tck.junit4.rule.DynamicPort;
+import org.mule.tck.junit4.rule.SystemProperty;
+
+import java.security.cert.CertPathValidatorException;
+import java.security.cert.CertificateRevokedException;
+
+import org.junit.Rule;
+
+public abstract class AbstractHttpTlsRevocationTestCase extends MuleArtifactFunctionalTestCase {
```

Comment:
Extend from `AbstractHttpTestCase` and add Story annotation if appropriate here.